devito/interfaces.py

```diff
@@ -18,8 +56,23 @@ def set_initializer(self, lambda_initializer):
         assert(callable(lambda_initializer))
         self.initializer = lambda_initializer
 
+    # function to generate a hased file name used for memmap file
+    def _generate_memmap_filenmae(self):
```

Comment:
Ok, I'm not sure what you mean by multiple "u"? `u = TimeData(...)` will always create one object with only one data buffer that you might want to spill and therefore require only one filepath. In fact the new symbolics infrastructure will effectively cache the data objects on their name so `TimeData(name='u', ...)` will always point to the same object and over-write the previous. So I'm really not convinced why you want to hash the file name.

Regarding the "hassle" of providing file names: That is why I want the symbol/object name, which the user always has to provide, to be part of the default path, so that most users won't need to overwrite the default. I do, however, want the option to manually overwrite this, because certain HPC clusters for example have special dedicated partitions for fast file IO, so an application developer might insert his preferred way to override the default paths; whether that is done via a prefix or a full path is your call.